api/src/main/java/com/velocitypowered/api/network/ProtocolState.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2024 Velocity Contributors
+ *
+ * The Velocity API is licensed under the terms of the MIT License. For more details,
+ * reference the LICENSE file in the api top-level directory.
+ */
+
+package com.velocitypowered.api.network;
+
+/**
+ * Representation of the state of the protocol
+ * in which a connection can be present.
+ */
+public enum ProtocolState {
```

Comment:
I'm not exactly fond of the documentation on this class, somebody with better tonguetangualation would probably be able to advise better than I can